### PR TITLE
Fix docstring in many different files

### LINKED ISSUE #112593: Fix docstring errors in nadam.py, radam.py, sgd.py, anomaly_mode.py, rprop.py, __init__.py, swa_utils.py, rmsprop.py, optimizer.py, lr_scheduler.py
Please fix the following issues. To test locally, run: 
 ```
pydocstyle path-to-file --count
```
Run this command before and after fixing the errors. This command prints out the number of errors at the end of the output. Paste the output to your PR description with the number before and after. 
- **File**: `torch/optim/lr_scheduler.py`, **Entity**: `_initial_step`, **Line**: 82, **Description**: First line should end with a period (not 'p')
- **File**: `torch/optim/lr_scheduler.py`, **Entity**: `state_dict`, **Line**: 88, **Description**: First line should be in imperative mood (perhaps 'Return', not 'Returns')
- **File**: `torch/optim/lr_scheduler.py`, **Entity**: `load_state_dict`, **Line**: 96, **Description**: First line should be in imperative mood (perhaps 'Load', not 'Loads')
- **File**: `torch/optim/lr_scheduler.py`, **Entity**: `get_last_lr`, **Line**: 105, **Description**: One-line docstring should fit on one line with quotes (found 2)
- **File**: `torch/optim/lr_scheduler.py`, **Entity**: `get_last_lr`, **Line**: 105, **Description**: No whitespaces allowed surrounding docstring text
- **File**: `torch/optim/lr_scheduler.py`, **Entity**: `print_lr`, **Line**: 114, **Description**: One-line docstring should fit on one line with quotes (found 2)
- **File**: `torch/optim/lr_scheduler.py`, **Entity**: `LambdaLR`, **Line**: 187, **Description**: 1 blank line required between summary line and description (found 0)
- **File**: `torch/optim/lr_scheduler.py`, **Entity**: `LambdaLR`, **Line**: 187, **Description**: First line should end with a period (not 'r')
- **File**: `torch/optim/lr_scheduler.py`, **Entity**: `state_dict`, **Line**: 224, **Description**: No blank lines allowed after function docstring (found 1)
- **File**: `torch/optim/lr_scheduler.py`, **Entity**: `state_dict`, **Line**: 224, **Description**: First line should be in imperative mood (perhaps 'Return', not 'Returns')
- **File**: `torch/optim/lr_scheduler.py`, **Entity**: `load_state_dict`, **Line**: 244, **Description**: No blank lines allowed after function docstring (found 1)
- **File**: `torch/optim/lr_scheduler.py`, **Entity**: `load_state_dict`, **Line**: 244, **Description**: First line should be in imperative mood (perhaps 'Load', not 'Loads')
- **File**: `torch/optim/lr_scheduler.py`, **Entity**: `MultiplicativeLR`, **Line**: 273, **Description**: 1 blank line required between summary line and description (found 0)
- **File**: `torch/optim/lr_scheduler.py`, **Entity**: `MultiplicativeLR`, **Line**: 273, **Description**: First line should end with a period (not 'n')
- **File**: `torch/optim/lr_scheduler.py`, **Entity**: `state_dict`, **Line**: 308, **Description**: First line should be in imperative mood (perhaps 'Return', not 'Returns')
- **File**: `torch/optim/lr_scheduler.py`, **Entity**: `load_state_dict`, **Line**: 325, **Description**: First line should be in imperative mood (perhaps 'Load', not 'Loads')
- **File**: `torch/optim/lr_scheduler.py`, **Entity**: `StepLR`, **Line**: 354, **Description**: 1 blank line required between summary line and description (found 0)
- **File**: `torch/optim/lr_scheduler.py`, **Entity**: `StepLR`, **Line**: 354, **Description**: First line should end with a period (not 'y')
- **File**: `torch/optim/lr_scheduler.py`, **Entity**: `MultiStepLR`, **Line**: 403, **Description**: 1 blank line required between summary line and description (found 0)
- **File**: `torch/optim/lr_scheduler.py`, **Entity**: `MultiStepLR`, **Line**: 403, **Description**: First line should end with a period (not 'e')
- **File**: `torch/optim/lr_scheduler.py`, **Entity**: `ConstantLR`, **Line**: 452, **Description**: 1 blank line required between summary line and description (found 0)
- **File**: `torch/optim/lr_scheduler.py`, **Entity**: `ConstantLR`, **Line**: 452, **Description**: First line should end with a period (not 'e')
- **File**: `torch/optim/lr_scheduler.py`, **Entity**: `LinearLR`, **Line**: 510, **Description**: 1 blank line required between summary line and description (found 0)
- **File**: `torch/optim/lr_scheduler.py`, **Entity**: `LinearLR`, **Line**: 510, **Description**: First line should end with a period (not 'l')
- **File**: `torch/optim/lr_scheduler.py`, **Entity**: `ExponentialLR`, **Line**: 578, **Description**: 1 blank line required between summary line and description (found 0)
- **File**: `torch/optim/lr_scheduler.py`, **Entity**: `SequentialLR`, **Line**: 609, **Description**: 1 blank line required between summary line and description (found 0)
- **File**: `torch/optim/lr_scheduler.py`, **Entity**: `SequentialLR`, **Line**: 609, **Description**: First line should end with a period (not 'g')
- **File**: `torch/optim/lr_scheduler.py`, **Entity**: `state_dict`, **Line**: 686, **Description**: First line should be in imperative mood (perhaps 'Return', not 'Returns')
- **File**: `torch/optim/lr_scheduler.py`, **Entity**: `load_state_dict`, **Line**: 701, **Description**: First line should be in imperative mood (perhaps 'Load', not 'Loads')
- **File**: `torch/optim/lr_scheduler.py`, **Entity**: `PolynomialLR`, **Line**: 718, **Description**: 1 blank line required after class docstring (found 0)
- **File**: `torch/optim/lr_scheduler.py`, **Entity**: `PolynomialLR`, **Line**: 718, **Description**: 1 blank line required between summary line and description (found 0)
- **File**: `torch/optim/lr_scheduler.py`, **Entity**: `PolynomialLR`, **Line**: 718, **Description**: First line should end with a period (not 'n')
- **File**: `torch/optim/lr_scheduler.py`, **Entity**: `CosineAnnealingLR`, **Line**: 768, **Description**: 1 blank line required between summary line and description (found 0)
- **File**: `torch/optim/lr_scheduler.py`, **Entity**: `CosineAnnealingLR`, **Line**: 768, **Description**: First line should end with a period (not 'g')
- **File**: `torch/optim/lr_scheduler.py`, **Entity**: `ChainedScheduler`, **Line**: 841, **Description**: 1 blank line required between summary line and description (found 0)
- **File**: `torch/optim/lr_scheduler.py`, **Entity**: `ChainedScheduler`, **Line**: 841, **Description**: First line should end with a period (not 'g')
- **File**: `torch/optim/lr_scheduler.py`, **Entity**: `state_dict`, **Line**: 882, **Description**: First line should be in imperative mood (perhaps 'Return', not 'Returns')
- **File**: `torch/optim/lr_scheduler.py`, **Entity**: `load_state_dict`, **Line**: 897, **Description**: First line should be in imperative mood (perhaps 'Load', not 'Loads')
- **File**: `torch/optim/lr_scheduler.py`, **Entity**: `ReduceLROnPlateau`, **Line**: 914, **Description**: 1 blank line required between summary line and description (found 0)
- **File**: `torch/optim/lr_scheduler.py`, **Entity**: `_reset`, **Line**: 1002, **Description**: First line should be in imperative mood (perhaps 'Reset', not 'Resets')
- **File**: `torch/optim/lr_scheduler.py`, **Entity**: `CyclicLR`, **Line**: 1088, **Description**: 1 blank line required between summary line and description (found 0)
- **File**: `torch/optim/lr_scheduler.py`, **Entity**: `CyclicLR`, **Line**: 1088, **Description**: First line should end with a period (not 'o')
- **File**: `torch/optim/lr_scheduler.py`, **Entity**: `get_lr`, **Line**: 1285, **Description**: No blank lines allowed after function docstring (found 1)
- **File**: `torch/optim/lr_scheduler.py`, **Entity**: `get_lr`, **Line**: 1285, **Description**: 1 blank line required between summary line and description (found 0)
- **File**: `torch/optim/lr_scheduler.py`, **Entity**: `get_lr`, **Line**: 1285, **Description**: First line should end with a period (not 's')
- **File**: `torch/optim/lr_scheduler.py`, **Entity**: `get_lr`, **Line**: 1285, **Description**: First line should be in imperative mood (perhaps 'Calculate', not 'Calculates')
- **File**: `torch/optim/lr_scheduler.py`, **Entity**: `CosineAnnealingWarmRestarts`, **Line**: 1339, **Description**: 1 blank line required between summary line and description (found 0)
- **File**: `torch/optim/lr_scheduler.py`, **Entity**: `CosineAnnealingWarmRestarts`, **Line**: 1339, **Description**: First line should end with a period (not 'g')
- **File**: `torch/optim/lr_scheduler.py`, **Entity**: `step`, **Line**: 1390, **Description**: No blank lines allowed after function docstring (found 1)
- **File**: `torch/optim/lr_scheduler.py`, **Entity**: `step`, **Line**: 1390, **Description**: First line should end with a period (not 'e')
- **File**: `torch/optim/lr_scheduler.py`, **Entity**: `OneCycleLR`, **Line**: 1464, **Description**: 1 blank line required after class docstring (found 0)
- **File**: `torch/optim/lr_scheduler.py`, **Entity**: `OneCycleLR`, **Line**: 1464, **Description**: 1 blank line required between summary line and description (found 0)
- **File**: `torch/optim/lr_scheduler.py`, **Entity**: `OneCycleLR`, **Line**: 1464, **Description**: First line should end with a period (not 'e')
- **File**: `torch/optim/lr_scheduler.py`, **Entity**: `_annealing_cos`, **Line**: 1695, **Description**: Use """triple double quotes""" (found "-quotes)
- **File**: `torch/optim/lr_scheduler.py`, **Entity**: `_annealing_linear`, **Line**: 1701, **Description**: Use """triple double quotes""" (found "-quotes)
- **File**: `torch/optim/nadam.py`, **Entity**: `step`, **Line**: 70, **Description**: First line should be in imperative mood (perhaps 'Perform', not 'Performs')
- **File**: `torch/optim/nadam.py`, **Entity**: `nadam`, **Line**: 176, **Description**: No blank lines allowed after function docstring (found 2)
- **File**: `torch/optim/optimizer.py`, **Entity**: `_RequiredParameter`, **Line**: 25, **Description**: 1 blank line required after class docstring (found 0)
- **File**: `torch/optim/optimizer.py`, **Entity**: `register_optimizer_step_pre_hook`, **Line**: 143, **Description**: 1 blank line required between summary line and description (found 0)
- **File**: `torch/optim/optimizer.py`, **Entity**: `register_optimizer_step_pre_hook`, **Line**: 143, **Description**: First line should end with a period (not 'g')
- **File**: `torch/optim/optimizer.py`, **Entity**: `register_optimizer_step_post_hook`, **Line**: 162, **Description**: 1 blank line required between summary line and description (found 0)
- **File**: `torch/optim/optimizer.py`, **Entity**: `register_optimizer_step_post_hook`, **Line**: 162, **Description**: First line should end with a period (not 'g')
- **File**: `torch/optim/optimizer.py`, **Entity**: `_group_tensors_by_device_and_dtype`, **Line**: 323, **Description**: 1 blank line required between summary line and description (found 0)
- **File**: `torch/optim/optimizer.py`, **Entity**: `_group_tensors_by_device_and_dtype`, **Line**: 323, **Description**: Multi-line docstring closing quotes should be on a separate line
- **File**: `torch/optim/optimizer.py`, **Entity**: `_group_tensors_by_device_and_dtype`, **Line**: 323, **Description**: First line should be in imperative mood (perhaps 'Group', not 'Groups')
- **File**: `torch/optim/optimizer.py`, **Entity**: `register_step_pre_hook`, **Line**: 338, **Description**: 1 blank line required between summary line and description (found 0)
- **File**: `torch/optim/optimizer.py`, **Entity**: `register_step_pre_hook`, **Line**: 338, **Description**: First line should end with a period (not 'e')
- **File**: `torch/optim/optimizer.py`, **Entity**: `register_step_post_hook`, **Line**: 360, **Description**: 1 blank line required between summary line and description (found 0)
- **File**: `torch/optim/optimizer.py`, **Entity**: `state_dict`, **Line**: 380, **Description**: First line should be in imperative mood (perhaps 'Return', not 'Returns')
- **File**: `torch/optim/optimizer.py`, **Entity**: `load_state_dict`, **Line**: 432, **Description**: First line should be in imperative mood (perhaps 'Load', not 'Loads')
- **File**: `torch/optim/optimizer.py`, **Entity**: `zero_grad`, **Line**: 488, **Description**: First line should be in imperative mood (perhaps 'Reset', not 'Resets')
- **File**: `torch/optim/optimizer.py`, **Entity**: `step`, **Line**: 529, **Description**: First line should be in imperative mood (perhaps 'Perform', not 'Performs')
- **File**: `torch/optim/optimizer.py`, **Entity**: `step`, **Line**: 529, **Description**: First line should not be the function's "signature"
- **File**: `torch/optim/radam.py`, **Entity**: `step`, **Line**: 84, **Description**: First line should be in imperative mood (perhaps 'Perform', not 'Performs')
- **File**: `torch/optim/radam.py`, **Entity**: `radam`, **Line**: 200, **Description**: No blank lines allowed after function docstring (found 1)
- **File**: `torch/optim/rmsprop.py`, **Entity**: `step`, **Line**: 97, **Description**: First line should be in imperative mood (perhaps 'Perform', not 'Performs')
- **File**: `torch/optim/rmsprop.py`, **Entity**: `rmsprop`, **Line**: 217, **Description**: No blank lines allowed after function docstring (found 1)
- **File**: `torch/optim/rmsprop.py`, **Entity**: `rmsprop`, **Line**: 217, **Description**: 1 blank line required between summary line and description (found 0)
- **File**: `torch/optim/rprop.py`, **Entity**: `step`, **Line**: 82, **Description**: First line should be in imperative mood (perhaps 'Perform', not 'Performs')
- **File**: `torch/optim/rprop.py`, **Entity**: `rprop`, **Line**: 188, **Description**: No blank lines allowed after function docstring (found 1)
- **File**: `torch/optim/sgd.py`, **Entity**: `step`, **Line**: 57, **Description**: First line should be in imperative mood (perhaps 'Perform', not 'Performs')
- **File**: `torch/optim/sgd.py`, **Entity**: `sgd`, **Line**: 200, **Description**: No blank lines allowed after function docstring (found 1)
- **File**: `torch/optim/swa_utils.py`, **Entity**: `AveragedModel`, **Line**: 67, **Description**: 1 blank line required after class docstring (found 0)
- **File**: `torch/optim/swa_utils.py`, **Entity**: `AveragedModel`, **Line**: 67, **Description**: 1 blank line required between summary line and description (found 0)
- **File**: `torch/optim/swa_utils.py`, **Entity**: `AveragedModel`, **Line**: 67, **Description**: First line should end with a period (not 'd')
- **File**: `torch/optim/swa_utils.py`, **Entity**: `update_bn`, **Line**: 227, **Description**: First line should be in imperative mood (perhaps 'Update', not 'Updates')
- **File**: `torch/optim/swa_utils.py`, **Entity**: `SWALR`, **Line**: 280, **Description**: 1 blank line required after class docstring (found 0)
- **File**: `torch/optim/_multi_tensor/__init__.py`, **Entity**: ``, **Line**: 1, **Description**: 1 blank line required between summary line and description (found 0)
- **File**: `torch/autograd/__init__.py`, **Entity**: ``, **Line**: 1, **Description**: 1 blank line required between summary line and description (found 0)
- **File**: `torch/autograd/__init__.py`, **Entity**: ``, **Line**: 1, **Description**: First line should end with a period (not 'c')
- **File**: `torch/autograd/__init__.py`, **Entity**: `backward`, **Line**: 118, **Description**: 1 blank line required between summary line and description (found 0)
- **File**: `torch/autograd/__init__.py`, **Entity**: `backward`, **Line**: 118, **Description**: First line should end with a period (not 'h')
- **File**: `torch/autograd/__init__.py`, **Entity**: `backward`, **Line**: 118, **Description**: First line should be in imperative mood (perhaps 'Compute', not 'Computes')
- **File**: `torch/autograd/__init__.py`, **Entity**: `grad`, **Line**: 219, **Description**: 1 blank line required between summary line and description (found 0)
- **File**: `torch/autograd/__init__.py`, **Entity**: `grad`, **Line**: 219, **Description**: First line should end with a period (not 'o')
- **File**: `torch/autograd/__init__.py`, **Entity**: `grad`, **Line**: 219, **Description**: First line should be in imperative mood (perhaps 'Compute', not 'Computes')
- **File**: `torch/autograd/anomaly_mode.py`, **Entity**: `detect_anomaly`, **Line**: 10, **Description**: No blank lines allowed between a section header and its content ('Example')

cc @brycebortree @carljparker

### PR DESCRIPTION
**Pull request by spzala**
This PR takes care of over 150 docstring errors in about a dozen files to reduce errors to 0. This can be verified by running, `pydocstyle path-to-file --count` as below:

**BEFORE the PR:** 
`pydocstyle torch/optim/lr_scheduler.py --count `
92 
`pydocstyle torch/optim/nadam.py --count`
6
`pydocstyle torch/optim/radam.py  --count`
6
`pydocstyle torch/optim/sgd.py --count`
6
`pydocstyle torch/autograd/anomaly_mode.py --count`
8
`pydocstyle torch/autograd/__init__.py --count`
9
`pydocstyle torch/optim/rprop.py --count`
5
 `pydocstyle torch/optim/_multi_tensor/__init__.py --count`
2
`pydocstyle torch/optim/swa_utils.py --count`
14
`pydocstyle torch/optim/rmsprop.py --count`
6
`pydocstyle torch/optim/optimizer.py --count`
29

**AFTER the PR:** 
`pydocstyle torch/optim/lr_scheduler.py --count `
0
`pydocstyle torch/optim/nadam.py --count `
0
`pydocstyle torch/optim/radam.py  --count`
0
`pydocstyle torch/optim/sgd.py --count`
0
`pydocstyle torch/autograd/anomaly_mode.py --count`
0
`pydocstyle torch/autograd/__init__.py --count`
0
`pydocstyle torch/optim/rprop.py --count`
0
`pydocstyle orch/optim/_multi_tensor/__init__.py --count`
0
`pydocstyle torch/optim/swa_utils.py --count`
0
`pydocstyle torch/optim/rmsprop.py --count`
0
`pydocstyle torch/optim/optimizer.py --count`
0

Fixes #112593

cc @svekars @brycebortree @vincentqb @jbschlosser @albanD @janeyx99 @crcrpar

### CODE REVIEW

**Comment by spzala** [comment]
cc @svekars

File: torch/autograd/anomaly_mode.py [hunk 1]
```diff
@@ -74,6 +74,7 @@ class detect_anomaly:
     """
 
     def __init__(self, check_nan=True) -> None:
+        """Init."""
```
Comment (soulitzer) [inline torch/autograd/anomaly_mode.py:77]: Hmmm this doesn't seem too useful lol, would prefer just to do some kind of way to skip the check if possible.

Comment (jbschlosser) [reply torch/autograd/anomaly_mode.py:77]: Yeah I agree with this!

Comment (svekars) [reply torch/autograd/anomaly_mode.py:77]: I believe to skip it should be `# noqa:  D412`

File: torch/autograd/anomaly_mode.py [hunk 2]
```diff
@@ -109,12 +112,15 @@ class set_detect_anomaly:
     """
 
     def __init__(self, mode: bool, check_nan: bool = True) -> None:
+        """Init."""
```
Comment (svekars) [inline torch/autograd/anomaly_mode.py:115]: Same as above

File: torch/autograd/anomaly_mode.py (hunk 1 above)
Comment (spzala) [reply torch/autograd/anomaly_mode.py:77]: @soulitzer @jbschlosser I agree too but it was purely added just to satisfy the docstring requirement from `pydocstyle` as I didn't find anyway to skip it. There is a general way to ignore checking for certain error code using `--ignore` flag when we run `pydocstyle` tool but that won't help much. I like the suggestion from @svekars and I will make changes accordingly. Thanks all for the review comments!!

Comment (spzala) [reply torch/autograd/anomaly_mode.py]: I have made updates. @svekars for the `__init__` it required to use error code D107 so I have added `# noqa: D107` and verified that it worked. Thanks so much again!

Comment (spzala) [reply torch/autograd/anomaly_mode.py]: Oh, I just noticed that I accidentally committed a submodule changes as well :( I will need to fix it. Thanks!

**Comment by soulitzer** [comment]
Thanks for the update, @spzala!
 
What do you think about just adding a skip for all the ones that did not have a docstring before as a first step. Then if there are particular ones you think would particularly benefit from having a docstring, we could add those as a follow up. That way things would be easier to review.

**Comment by spzala** [comment]
> Thanks for the update, @spzala!
> 
> What do you think about just adding a skip for all the ones that did not have a docstring before as a first step. Then if there are particular ones you think would particularly benefit from having a docstring, we could add those as a follow up. That way things would be easier to review.

@soulitzer thank you and sure, that sounds good!

**Comment by spzala** [comment]
WIP

**Comment by spzala** [comment]
> > Thanks for the update, @spzala!
> > What do you think about just adding a skip for all the ones that did not have a docstring before as a first step. Then if there are particular ones you think would particularly benefit from having a docstring, we could add those as a follow up. That way things would be easier to review.
> 
> @soulitzer thank you and sure, that sounds good!

@soulitzer @svekars the PR is ready for your review. I have skipped many obvious places where I think docstring is not needed but did make changes where I thought we should make changes per the issue this PR is created for. Please let me know your comments. I will be happy to make more changes per your feedback. Thanks!

File: torch/autograd/__init__.py [hunk 3]
```diff
@@ -477,6 +475,7 @@ def _is_checkpoint_valid():
 
 
 def variable(*args, **kwargs):
+    """Provide runtime error for deprecated autograd variable."""
```
Comment (soulitzer) [inline torch/autograd/__init__.py:478]: Could we turn this into a skip instead?

**Comment by soulitzer** [comment]
autograd parts look OK to me, added one comment
@janeyx99 should take a closer look at the optim parts

File: torch/autograd/__init__.py (hunk 3 above)
Comment (spzala) [reply torch/autograd/__init__.py:478]: @soulitzer sure, that sounds good. Thanks!

File: torch/optim/lr_scheduler.py [hunk 4]
```diff
@@ -139,12 +144,12 @@ def wrapper(*args, **kwargs):
         self._initial_step()
 
     def _initial_step(self):
-        """Initialize step counts and performs a step"""
+        """Initialize step counts and performs a step."""
```
Comment (janeyx99) [inline torch/optim/lr_scheduler.py:147]:
```suggestion
        """Initialize step counts and perform a step."""
```

File: torch/optim/lr_scheduler.py [hunk 5]
```diff
@@ -154,7 +159,7 @@ def state_dict(self):
         }
 
     def load_state_dict(self, state_dict: Dict[str, Any]):
-        """Loads the schedulers state.
+        """Load the schedulers state.
```
Comment (janeyx99) [inline torch/optim/lr_scheduler.py:162]:
```suggestion
        """Load the scheduler's state.
```

File: torch/optim/lr_scheduler.py [hunk 6]
```diff
@@ -351,15 +358,14 @@ def state_dict(self):
         return state_dict
 
     def load_state_dict(self, state_dict):
-        """Loads the schedulers state.
+        """Load the schedulers state.
```
Comment (janeyx99) [inline torch/optim/lr_scheduler.py:361]:
```suggestion
        """Load the scheduler's state.
```

File: torch/optim/lr_scheduler.py [hunk 7]
```diff
@@ -380,8 +387,9 @@ def get_lr(self):
 
 
 class MultiplicativeLR(LRScheduler):
-    """Multiply the learning rate of each parameter group by the factor given
-    in the specified function. When last_epoch=-1, sets initial lr as lr.
+    """Multiply the learning rate of each parameter group by the factor given in the specified function.
+
+    When last_epoch=-1, sets initial lr as lr.
```
Comment (janeyx99) [inline torch/optim/lr_scheduler.py:392]:
```suggestion
    When last_epoch=-1, set initial lr as lr.
```

File: torch/optim/lr_scheduler.py [hunk 8]
```diff
@@ -448,7 +456,7 @@ def state_dict(self):
         return state_dict
 
     def load_state_dict(self, state_dict):
-        """Loads the schedulers state.
+        """Load the schedulers state.
```
Comment (janeyx99) [inline torch/optim/lr_scheduler.py:459]:
```suggestion
        """Load the scheduler's state.
```

File: torch/optim/lr_scheduler.py [hunk 9]
```diff
@@ -671,8 +685,9 @@ def _get_closed_form_lr(self):
 
 
 class LinearLR(LRScheduler):
-    """Decays the learning rate of each parameter group by linearly changing small
-    multiplicative factor until the number of epoch reaches a pre-defined milestone: total_iters.
+    """Decays the learning rate of each parameter group by linearly changing small multiplicative factor.
+
+    The multiplicatino is done until the number of epoch reaches a pre-defined milestone: total_iters.
```
Comment (janeyx99) [inline torch/optim/lr_scheduler.py:690]:
```suggestion
    The multiplication is done until the number of epoch reaches a pre-defined milestone: total_iters.
```

File: torch/optim/lr_scheduler.py [hunk 10]
```diff
@@ -803,7 +821,9 @@ def _get_closed_form_lr(self):
 
 
 class SequentialLR(LRScheduler):
-    """Receives the list of schedulers that is expected to be called sequentially during
+    """Receives the list of schedulers.
```
Comment (janeyx99) [inline torch/optim/lr_scheduler.py:824]:
```suggestion
    """Contains a list of schedulers expected to be called sequentially during the optimization process.
```

File: torch/optim/lr_scheduler.py [hunk 11]
```diff
@@ -803,7 +821,9 @@ def _get_closed_form_lr(self):
...
+
+    These schedulers are expected to be called sequentially during
     optimization process and milestone points that provides exact intervals to reflect
     which scheduler is supposed to be called at a given epoch.
```
Comment (janeyx99) [inline torch/optim/lr_scheduler.py:828]:
```suggestion
    Specifically, the schedulers will be called according to the milestone points, which should provide exact
    intervals by which each scheduler should be called at a given epoch.
```

File: torch/optim/lr_scheduler.py [hunk 12]
```diff
@@ -922,7 +943,7 @@ def state_dict(self):
         return state_dict
 
     def load_state_dict(self, state_dict):
-        """Loads the schedulers state.
+        """Load the schedulers state.
```
Comment (janeyx99) [inline torch/optim/lr_scheduler.py:946]:
```suggestion
        """Load the scheduler's state.
```

File: torch/optim/lr_scheduler.py [hunk 13]
```diff
@@ -1054,12 +1078,13 @@ def __init__(
...
         last_epoch=-1,
         verbose="deprecated",
-    ):
+    ):  # noqa: D107
         self.T_max = T_max
         self.eta_min = eta_min
         super().__init__(optimizer, last_epoch, verbose)
 
     def get_lr(self):
+        """Receive the learning rate of each parameter group."""
```
Comment (janeyx99) [inline torch/optim/lr_scheduler.py:1087]:
```suggestion
        """Retrieve the learning rate of each parameter group."""
```

File: torch/optim/lr_scheduler.py [hunk 14]
```diff
@@ -1097,9 +1122,10 @@ def _get_closed_form_lr(self):
 
 
 class ChainedScheduler(LRScheduler):
-    """Chains list of learning rate schedulers. It takes a sequence of chainable learning
-    rate schedulers and performs consecutive step() functions belonging to them by just
-    one call.
+    """Chains list of learning rate schedulers.
```
Comment (janeyx99) [inline torch/optim/lr_scheduler.py:1125]:
```suggestion
    """Chains a list of learning rate schedulers.
```

File: torch/optim/lr_scheduler.py [hunk 15]
```diff
@@ -1097,9 +1122,10 @@ def _get_closed_form_lr(self):
...
+
+    It takes a sequence of chainable learning rate schedulers and performs consecutive step()
+    functions belonging to them by just one call.
```
Comment (janeyx99) [inline torch/optim/lr_scheduler.py:1128]:
```suggestion
    Takes in a sequence of chainable learning rate schedulers and calls their step() functions
    consecutively in just one call to step().
```

File: torch/optim/lr_scheduler.py [hunk 16]
```diff
@@ -1181,7 +1208,7 @@ def state_dict(self):
         return state_dict
 
     def load_state_dict(self, state_dict):
-        """Loads the schedulers state.
+        """Load the schedulers state.
```
Comment (janeyx99) [inline torch/optim/lr_scheduler.py:1211]:
```suggestion
        """Load the scheduler's state.
```

File: torch/optim/lr_scheduler.py [hunk 17]
```diff
@@ -1308,12 +1336,13 @@ def __init__(
...
 
     def _reset(self):
-        """Resets num_bad_epochs counter and cooldown counter."""
+        """Reset num_bad_epochs counter and cooldown counter."""
         self.best = self.mode_worse
         self.cooldown_counter = 0
         self.num_bad_epochs = 0
 
     def step(self, metrics: SupportsFloat, epoch=None):  # type: ignore[override]
+        """Receive learning rates for parameter groups."""
```
Comment (janeyx99) [inline torch/optim/lr_scheduler.py:1345]:
```suggestion
        """Perform a step."""
```

File: torch/optim/lr_scheduler.py [hunk 18]
```diff
@@ -1380,23 +1409,24 @@ def _init_is_better(self, mode, threshold, threshold_mode):
...
         self.threshold_mode = threshold_mode
 
-    def state_dict(self):
+    def state_dict(self):  # noqa: D102
         return {
             key: value for key, value in self.__dict__.items() if key != "optimizer"
         }
 
     def load_state_dict(self, state_dict):
+        """Update mode and threshold values."""
```
Comment (janeyx99) [inline torch/optim/lr_scheduler.py:1418]:
```suggestion
        """Load the scheduler's state"""
```

File: torch/optim/lr_scheduler.py [hunk 19]
```diff
@@ -1585,6 +1615,7 @@ def _init_scale_fn(self):
             self.scale_mode = "iterations"
 
     def scale_fn(self, x) -> float:
+        """Receive scaling policy."""
```
Comment (janeyx99) [inline torch/optim/lr_scheduler.py:1618]:
```suggestion
        """Get the scaling policy."""
```

File: torch/optim/lr_scheduler.py [hunk 20]
```diff
@@ -1650,6 +1681,7 @@ def get_lr(self):
         return lrs
 
     def state_dict(self):
+        """Receive initialized or updated scale function."""
```
Comment (janeyx99) [inline torch/optim/lr_scheduler.py:1684]: I think your docs here is wrong.

File: torch/optim/lr_scheduler.py [hunk 21]
```diff
@@ -1664,6 +1696,7 @@ def state_dict(self):
         return state
 
     def load_state_dict(self, state_dict):
+        """Update scale function if found, initialize otherwise."""
```
Comment (janeyx99) [inline torch/optim/lr_scheduler.py:1699]: also wrong here.

File: torch/optim/nadam.py [hunk 22]
```diff
@@ -1,4 +1,5 @@
 # mypy: allow-untyped-defs
+r"""NAdam algorithm."""
```
Comment (janeyx99) [inline torch/optim/nadam.py:2]: how does this render? where does this render to?

File: torch/optim/nadam.py [hunk 23]
```diff
@@ -25,6 +26,8 @@
 
 
 class NAdam(Optimizer):
+    r"""Implementation for the NAdam algorithm."""
```
Comment (janeyx99) [inline torch/optim/nadam.py:29]: we intentionally set the doc for this class later to give ourselves more freedom in using fstrings.

Comment (janeyx99) [reply torch/optim/nadam.py:29]: This is true for all other optimizer files as well (other than the base)

File: torch/optim/optimizer.py [hunk 24]
```diff
@@ -754,9 +758,11 @@ def register_load_state_dict_pre_hook(
         hook: Callable[["Optimizer", StateDict], Optional[StateDict]],
         prepend: bool = False,
     ) -> RemovableHandle:
-        r"""Register a load_state_dict pre-hook which will be called before
-        :meth:`~torch.optim.Optimizer.load_state_dict` is called. It should have the
-        following signature::
+        r"""Register a load_state_dict pre-hook.
```
Comment (janeyx99) [inline torch/optim/optimizer.py:761]: why is this change necessary? The first and second separated line here is a better first line together.

File: torch/optim/nadam.py (hunk 23 above)
Comment (janeyx99) [reply torch/optim/nadam.py:29]: I will wait for a response on this before reviewing the rest of the optimizers.

**Review by janeyx99 — COMMENTED** [review]
Thanks for taking this big task on! 

It would be better if you could split this PR into different topics instead of lumping into one PR. One for autograd, one for LRScheduler, and one for optimizer. The first two seem almost there, but the optimizer doc strings changes will have major undesirable implications.

**Comment by spzala** [comment]
> Thanks for taking this big task on!
> 
> It would be better if you could split this PR into different topics instead of lumping into one PR. One for autograd, one for LRScheduler, and one for optimizer. The first two seem almost there, but the optimizer doc strings changes will have major undesirable implications.

Hi @janeyx99 thanks so much for the review and suggestions! Yes, I initially thought so but I wasn't sure considering Docathon, it will definitely make it easy to review three separate PRs. I will work on splitting this PR and address your comments. Thanks!

File: torch/optim/lr_scheduler.py (hunk 9 above)
Comment (spzala) [reply torch/optim/lr_scheduler.py:690]: Good eye :) @janeyx99 Thanks!

**Comment by svekars** [comment]
we should close this PR since these fixes are implemented in two other PRs.

File: torch/optim/nadam.py (hunk 22 above)
Comment (spzala) [reply torch/optim/nadam.py:2]: This seems ignored in the documentation (unless I missed anything) but takes care of docstring error.

File: torch/optim/nadam.py (hunk 23 above)
Comment (spzala) [reply torch/optim/nadam.py:29]: Thanks @janeyx99 I have ignored it in this Optim specific changes, https://github.com/pytorch/pytorch/pull/129086
